keps/sig-contributor-experience/1553-issue-triage/README.md

```diff
@@ -0,0 +1,225 @@
+- [690 are categorized as `lifecycle/frozen`](https://github.com/kubernetes/kubernetes/labels/lifecycle%2Ffrozen)
+
+This makes for about 20% of open issues/PRs that are in some state of staleness.
+If we consider items marked as `lifecycle/frozen`, then we're looking at around
+42% of issues/PRs that could potentially require attention.
+
+### Goals
+
+See [User Stories](#user-stories).
+
+<unresolved>
+
+### Non-Goals
+
+- Prescribing an issue triage workflow for all projects within Kubernetes orgs
+
+</unresolved>
+
+## Proposal
+
+### User Stories
+
+#### Group Leads
+
+As a SIG Chair/Technical Lead, WG/UG organizer, or subproject owner, I want to
+be able to easily triage my group's backlog of open issues and PRs to accelerate
+merge velocity and issue resolution.
+
+#### Reviewers and Approvers
+
+As a reviewer/approver, I want a simple system to categorize issues/PRs in my
+purview, so that I can prioritize what to review first.
+
+#### Contributors
+
+As a contributor, I want to be able to submit issues or PRs and:
+
+- understand how the community works to address new submissions
+- have some assurance that they will be routed to the correct group
+- have them addressed in a timely manner
+
+<unresolved>
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they releate.
+-->
+
+</unresolved>
+
+<unresolved>
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+</unresolved>
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Graduation Criteria
+
+<!--
+**Note:** *Not required until targeted at a release.*
+-->
+
+_Not required in `provisional` state._
+
+### Phase 0
+
+#### needs-triage and lifecycle/ready labels
+
+We currently have a Prow plugin called `require-matching-label`, which requires
+specific labels to be set on a issue or pull request.
+
+Examples in [kubernetes/kubernetes] include:
+
+- `needs-kind` (needs a `kind/foo` label)
+- `needs-priority` (needs a `priority/bar` label)
+- `needs-sig` (needs a `sig/baz` label)
+
+Here we propose [introducing two new labels](https://github.com/kubernetes/test-infra/pull/16298):
+
+- `needs-triage`
+- `lifecycle/ready`
```

Comment:
Further down there is this comment:
> As for the lifecycle/active label, I don't like it.
> I think if an issue is assigned, it should be assumed to be active.

I think this may be a contributing factor to haring lots of open issues and PRs.
It is very easy for people to volunteer and assign themselves issues or PRs, or for other people to do it while asking for a review of some sort.
If people see others assigned it could be that it makes it less likely for other to jump in (i.e., may be the assignees are working on issue X offline/in slack/in some un-linked PR).
If `/lifecycle active` was used throughout, fejta bot would help differentiate between things that are indeed active and those in which people were assigned or assigned themselves and forgot (were not able to carry on with the work for one reason or another).

The difference between `/lifecycle active` and `/lifecycle ready` can be a very subtle one.
I don't know how much info people would be able to gather from `/lifecycle active` that wouldn't be present from `/lifecycle ready` and vice versa (plus `/lifecycle active` already exists).
Label meaning should be as obvious as possible to facilitate adoption.

`/lifecycle active` already exists, if we could enforce it with the default addition of "`needs-lifecycle`" to every issue/PR, then with the help of fejta bot we could see if the issue/pr is indeed being worked on or if people are assigned to it but not working on it.
Just checking if people are assigned doesn't provide for the same workflow as one would have to explicitly open up the issue and read the timeline of events (compared to querying issues based on `-label:lifecycle/active`).

Thus what would you think about adding a `needs-lifecycle` and adding something like `/lifecycle proposed`.
The former to signal that the issue has not been completely triaged, the latter to "complete" the user experience in the sense that people can add it to issues to fulfill the need for a lifecycle label but still communicate that no one is actively working on it for a reason or another (the reason would be communicated in combination with priority and kind labels).

Reply:
I lean towards "less is more".  I think there are already too many labels, and it just won't get used well.

Reply:
I agree with the previous comments, and would not add labels when they don't add that much distinctive information when compared to existing ones (like `lifecycle/ready`, as argued by Jorge), but I think there is a strong case for adding the `needs-triage` (or `needs-lifecycle`) label for the proposed use case.

Reply:
+1 on needs-triage. 

We have a biweekly (as in 2x a week) bug-scrub and we keep track of our triaged PRs and issues manually in a google doc. There are some short-comings to this, namely:

1. PRs which are initially marked as WIP tend to get overlooked, since we only ever keep a pointer to the most recently triaged PR. 
2. PRs created before our meeting but only gain the sig label after our meeting also get overlooked. 
3. We don't have a good mechanism for ever revisiting something.

For our use-case, we just need a label to allow us to do better filtering and needs-triage would be sufficient for 95%+ of what we want to do.

Reply:
From a UX standpoint, I'd prefer the opposite of `needs-triage` to be a `triage/` class label.

For provisional, can we just make this a "placeholder"? Then we can bikeshed on the specifics of labels prior to implementation. However, the goal of a default "needs-triage" state, and then a command/label applied to negate it and indicate it's ready for someone to work on LGTM.